msal/src/Microsoft.Identity.Client/TokenCache.cs

```diff
@@ -216,6 +219,29 @@ internal void OnBeforeWrite(TokenCacheNotificationArgs args)
                 }
             }
         }
+
+        /// <summary>
+        /// This method ensures the tenant id is set on the id token
+        /// The B2C scenario does not return the tenant id within the id token at the moment.
+        /// As a fall back, the tenant id on the id Token is set to the tenant id that is in the B2C authority originally passed in by the user.
+        /// </summary>
+        /// <param name="requestParams">Request parameters set by the user</param>
+        /// <param name="idToken">iD token returned from STS</param>
+        private void EnsureTenantIDIsSpecifiedForB2CAuthority(AuthenticationRequestParameters requestParams, IdToken idToken)
+        {
+            if (!string.IsNullOrEmpty(idToken.TenantId) || requestParams.Authority.AuthorityType != Core.Instance.AuthorityType.B2C)
+            {
+                return;
+            }
+            else
+            {
```

Comment:
remove. otherwise you are not reducing nesting. Also it's not required because in the other case you have already returned